src/Strategy/NuGet/ProjectAssetsJson.hs

```diff
@@ -0,0 +1,93 @@
+module Strategy.NuGet.ProjectAssetsJson
+  ( discover
+  , strategy
+  , buildGraph
+  , analyze
+
+  , ProjectAssetsJson(..)
+  ) where
+
+import Prologue
+
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+import           Polysemy
+import           Polysemy.Input
+import           Polysemy.Output
+
+import           DepTypes
+import           Discovery.Walk
+import           Effect.ReadFS
+import           Graphing (Graphing, unfold)
+import           Types
+
+discover :: Discover
+discover = Discover
+  { discoverName = "projectassetsjson"
+  , discoverFunc = discover'
+  }
+
+discover' :: Members '[Embed IO, Output ConfiguredStrategy] r => Path Abs Dir -> Sem r ()
+discover' = walk $ \_ subdirs files -> do
+  case find (\f -> fileName f == "project.assets.json") files of
+    Just file -> output (configure file)
+    Nothing -> pure ()
+
+  walkSkipNamed ["node_modules/"] subdirs
+
+strategy :: Strategy BasicFileOpts
+strategy = Strategy
+  { strategyName = "nuget-projectassetsjson"
+  , strategyAnalyze = \opts -> analyze
+      & fileInputJson @ProjectAssetsJson (targetFile opts)
+  , strategyModule = parent. targetFile
+  , strategyOptimal = NotOptimal
+  , strategyComplete = NotComplete
+  }
+
+data ProjectAssetsJson = ProjectAssetsJson
+  { targets     :: M.Map Text (M.Map Text NugetDep)
+  } deriving Show
+
+instance FromJSON ProjectAssetsJson where
+  parseJSON = withObject "ProjectAssetsJson" $ \obj ->
+    ProjectAssetsJson <$> obj .: "targets"
+
+data NugetDep = NugetDep
+  { depType    :: Text
+  , deepDeps   :: Maybe (M.Map Text Text)
+  } deriving Show
+   
+data CompleteNugetDep = CompleteNugetDep
+  { depName            :: Text
+  , depVersion         :: Text
+  , completeDepType    :: Text
+  , completeDeepDeps   :: Maybe (M.Map Text Text)
+  } deriving Show
+
+instance FromJSON NugetDep where
+  parseJSON = withObject "Dependency" $ \obj ->
+    NugetDep <$> obj .: "type"
+             <*> obj .:? "dependencies"
+
+analyze :: Member (Input ProjectAssetsJson) r => Sem r (Graphing Dependency)
+analyze = buildGraph <$> input
+
+buildGraph :: ProjectAssetsJson -> Graphing Dependency
+buildGraph project = unfold direct deepList toDependency
+    where
+    direct = concat $ (\(_,a) -> ((\(x,y) -> CompleteNugetDep (head $ T.splitOn "/" x) (last $ T.splitOn "/" x) (depType y) (deepDeps y)) <$> (M.toList a))) <$> (M.toList (targets project))
```

Comment:
`concat $ f <$> xs`
is the same as
`concat (map f xs)`
is the same as
`concatMap f xs`
(or `f =<< xs`, but `concatMap` is a clearer name)

let's use `concatMap` here

Reply:
`head` and `last` are problematic: they're "partial" functions (i.e. they can throw runtime exceptions) when the list is empty:
```hs
head :: [a] -> a
head (x:_) = x
head _ = error "empty list"

last :: [a] -> a
last [x] = x
last (_:xs) = last xs
last _ = error "empty list"
```

unfortunately the prelude contains a bunch of these partial functions.

we do have a safer alternative, though: pattern matching!
```hs
case T.splitOn "/" depString of
    [name,ver] -> undefined -- we know we have two parts
    _ -> undefined -- and this forces us to consider the empty case (or explicitly ignore it)
```